setup.py

```diff
@@ -77,12 +77,6 @@ def run(self):
             if os.path.isdir(e):
                 shutil.rmtree(e)
 
-        self.clean_optional_plugins()
-
-    @staticmethod
-    def clean_optional_plugins():
-        walk_plugins_setup_py(["clean", "--all"])
-
```

Comment:
Maybe I'm missing some bit. But this is changing the current behavior we have: If we have a `build/lib` folder inside `optional_plugins/html` for instance, the current "main clean" is removing those folders as well. But if do the same with this PR changes, `optional_plugins/*/build` will be not removed.

Did I missed something? Or are you proposing a change on the current behavior?

Reply:
@beraldoleal  Unless I'm mistaken, none of the installation scenarios we use in Makefile  will create a `build/lib`  under `optional_plugins/*/build`. If it does, this could be handled better when we switch from `setup.py` to `setup.cfg`.

Reply:
"build" it was just an example, we generate other folders/files there.

Also should we limit us only by the Makefile targets? For instance, some might need to run `python3 setup.py develop|build` against a specific plugin or even using our global `python3 setup.py plugin -i html`, so yes, we don't have Makefile targets, but we still make available some targets that might generate those folders.

I find useful to have a general "clean" that will clean everything recursively, but I'm open to change our current behavior, if it makes sense. We just need to stress the impact that this will cause to the contributors.

Reply:
@beraldoleal 

I think you're mixing the clean command from `setuptools` with the one from `distutils`.

If you clone the current avocado branch master and `cd`  to `optional_plugins/robot`, then run `python3 setup.py develop` or/and  `python3 setup.py build`  some files will be generated, indeed. If then you run  `python3 setup.py clean` nothing will be removed.  The removed calls doesn't remove anything at this moment and
we'll have to address this, specially if we split out the plugins. As I mentioned, this can be solved later.

The reason why 'python setup.py clean --all' shows a lot of output now removing `build/*`  directories and is probably misleading here, is because it's run from the main setup.py that have already imported `distutils`. However, it still does nothing because the individual targets are already emptied (eggs removed) when the local `python3 setup.py clean` is run.

You mentioned that an user might use directly the main `python3 setup.py build` to install avocado, agreed, but in this case the user won't install the plugins so the  'python setup.py clean --all' would still be useless.

Hope this make things more clear.

Reply:
> I think you're mixing the clean command from setuptools with the one from distutils.

I'm aware that setuptools clean doesn't have the `--all` option. But in terms of functionality we are changing the behavior here (see example bellow).

> If you clone the current avocado branch master and cd to optional_plugins/robot, then run python3 setup.py develop or/and python3 setup.py build some files will be generated, indeed. If then you run python3 setup.py clean nothing will be removed.

Indeed, but  appending `--all` it will remove things.  And with this PR we are leaving things behind.

> ...
> You mentioned that an user might use directly the main python3 setup.py build to install avocado, agreed, but in this case the user won't install the plugins so the python setup.py clean --all that wouldo the local calls in every plugin would still be useless.

Before this PR:

```
cd optional_plugins/robot/
python3 setup.py build
cd ../../
python3 setup.py clean --all
ls -lad optional_plulgins/robot/build
ls: cannot access 'optional_plugins/robot/build/': No such file or directory
```

After this PR:

```
cd optional_plugins/robot/
python3 setup.py build
cd ../../
python3 setup.py clean # since we removed --all here, I don't have another choice.
ls -lad optional_plulgins/robot/build
drwxrwxr-x. 3 local local 4096 Jan 19 13:20 optional_plugins/robot/build/
```
As I said, I'm just thinking in terms of functionality, this Patch is changing the current behavior. If this *really* makes sense, then we need at least to make it clear for contributors. By changing the tool we could avoid changing the behavior. Maybe `--all` it is not the proper option for that, and a new must be introduced.

Sorry if it is too much noise, by I'm still missing some gaps here.

Reply:
Ok, I see your point of view better now. I still think this is postponing a problem we'll have to handle in the future because the command clean seems to be doomed to dissapear from setuptools (see references below)
This is not high priority right now, so I can drop the second commit 025e52ae83bf0ea01628ba04bfcc7d24deb2a9b4 and we'll revisit this when migrating to `setup.cfg`

References:
* clean is not even mentioned at the summary table at https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html#summary 
my guess is people are just using git clean these days
* this issue has been open for some time without answer from upstream https://github.com/pypa/setuptools/issues/1347

Reply:
@ana I agree we have to handle with this problem soon, but in meanwhile I think it is a good idea to remove the commit as you suggested.

Reply:
> @ana I agree we have to handle with this problem soon, but in meanwhile I think it is a good idea to remove the commit as you suggested.

Not a good idea. Now  `--all` doesn't work because as mentioned above,  the clean command from setuptools  is not the same than the one from distutils-

Reply:
I was just testing and realized that. :(

Reply:
> I was just testing and realized that. :( How about renaming --all for something else and implementing it the way it is working today?

I'd rather drop the PR, this is going away anyway in the nearby future.

Reply:
I went through the options (and the references) and I don't understand why we can't extend the `Clean` command on the primary `setup.py` to also cover our non-standard plugin layout.  We already clean files that are non-standard such as the auto-generated `.rst` files.

I understand this is not perfect and the plugin's `setup.py` themselves should have the capabilities of cleanup, but I think letting Avocado's main `setup.py` handle it is a compromise solution we can live with because it doesn't regress in functionality.

Reply:
@clebergnu if I understood your comment properly, that's the current behavior before this PR and this PR doesn't change it. The main setup.py cleans the full tree.

Reply:
@clebergnu yes, if you're using commands from one setup.py (in the plugin) and then from the setup.py in  root directory, the behavior is like that. I thought you were talking about the main setup.py all the time...

I'm personally using `git clean` all the time so I'm closing this issue without plans to revisit it.